language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/SortingUtil.java

```diff
@@ -276,6 +277,20 @@ public static String genSortTextByAssignability(BallerinaCompletionContext conte
      * @return True if assignable
      */
     public static boolean isCompletionItemAssignable(LSCompletionItem completionItem, TypeSymbol typeSymbol) {
+        if (typeSymbol.typeKind() == TypeDescKind.TYPEDESC && completionItem.getType() == SYMBOL) {
+            Optional<Symbol> optionalSymbol = ((SymbolCompletionItem) completionItem).getSymbol();
+            if (optionalSymbol.isPresent()) {
+                if (optionalSymbol.get().kind() == SymbolKind.TYPE_DEFINITION
+                        || optionalSymbol.get().kind() == SymbolKind.TYPE) {
+                    TypeDescTypeSymbol typeDescTypeSymbol = (TypeDescTypeSymbol) typeSymbol;
+                    if (typeDescTypeSymbol.typeParameter().isPresent()) {
+                        Optional<TypeSymbol> optionalTypeSymbol = getSymbolFromCompletionItem(completionItem);
+                        return optionalTypeSymbol.isPresent()
+                                && optionalTypeSymbol.get().subtypeOf(typeDescTypeSymbol.typeParameter().get());
```

Comment:
Better if we can extract the logic of obtaining the type-parameter in to a separate method

Reply:
Do you mean like a utility. Where we pass a type symbol and if it has a type-parameter it will return it?

Reply:
```
public static Optional<TypeSymbol> getTypeParameterFromTypeSymbol(TypeSymbol typeSymbol) {
        Optional<TypeSymbol> optionalTypeSymbol;
        switch (typeSymbol.typeKind()) {
            case TYPEDESC:
                optionalTypeSymbol = ((TypeDescTypeSymbol) typeSymbol).typeParameter();
                break;
            case FUTURE:
                optionalTypeSymbol = ((FutureTypeSymbol)typeSymbol).typeParameter();
                break;
            default:
                optionalTypeSymbol = Optional.empty();
                break;
        }
        return optionalTypeSymbol;
    }
```

Reply:
> ```
> public static Optional<TypeSymbol> getTypeParameterFromTypeSymbol(TypeSymbol typeSymbol) {
>         Optional<TypeSymbol> optionalTypeSymbol;
>         switch (typeSymbol.typeKind()) {
>             case TYPEDESC:
>                 optionalTypeSymbol = ((TypeDescTypeSymbol) typeSymbol).typeParameter();
>                 break;
>             case FUTURE:
>                 optionalTypeSymbol = ((FutureTypeSymbol)typeSymbol).typeParameter();
>                 break;
>             default:
>                 optionalTypeSymbol = Optional.empty();
>                 break;
>         }
>         return optionalTypeSymbol;
>     }
> ```

Exactly. We can make this private and place in the same class (SortingUtil)